### PR TITLE
Upgrade cfitsio to 3.49

### PR DESCRIPTION
**Pull request by erykoff**
This upgrade primarily fixes a bug in cfitsio related to lossless GZIP compressed floating-point images.

### CODE REVIEW

**Comment by beckermr** [comment]
Can we add a test that passes with this version but didn't with the last one? That way we can tell in the future if the code regresses.

File: cfitsio3490/docs/changes.txt [hunk 1]
```diff
@@ -1,4 +1,58 @@
                    Log of Changes Made to CFITSIO
+		   
+Version 3.49 - Aug 2020
+
+  - Fix to imcompress.c.  It now turns off quantization if ZSCALE
+    and ZZERO columns are missing.  Treatment will be the same as
+    if ZQUANTIZ were set to 'NONE', even if ZQUANTIZ is present
+    and set to something else.
```
Comment (beckermr) [inline cfitsio3490/docs/changes.txt:8]: @esheldon This is jogging my memory. Did you code around this in a recent bug fix?

Comment (erykoff) [reply cfitsio3490/docs/changes.txt:8]: We recently added support for writing lossless compressed floating point images with setting the quantiz level to 0. And these files could be read by fitsio or astropy. But similar files written by astropy couldn’t be read by c/fitsio because of this bug. (I had thought it was a bug in astropy but it was this bug in cfitsio)

**Comment by erykoff** [comment]
To add a test for this we would need a file written by astropy (or have astropy as a dependence for this test which we probably don’t want). It could be a very small file and if that’s ok I can add a test.

**Comment by beckermr** [comment]
SGTM!

**Comment by erykoff** [comment]
Gah.  Clearly what I did to add the data file didn't work on travis, though it worked locally.  Any idea what I did wrong?

**Comment by beckermr** [comment]
add the file to the manifest.in

**Comment by erykoff** [comment]
...and now it's reading None, I have no idea.

**Comment by beckermr** [comment]
It appears the test data was written into extension 1.

```
(Pdb++) fits.hdu_list
[
  file: /Users/beckermr/Desktop/fitsio/fitsio/test_images/test_gzip_compressed_image.fits.fz
  extension: 0
  type: IMAGE_HDU
  image info:
    data type: u1
    dims: [], 
  file: /Users/beckermr/Desktop/fitsio/fitsio/test_images/test_gzip_compressed_image.fits.fz
  extension: 1
  type: IMAGE_HDU
  extname: COMPRESSED_IMAGE
  image info:
    compression: GZIP_2
    data type: f8
    dims: [10,10]]
(Pdb++) fitsio.read("/Users/beckermr/Desktop/fitsio/fitsio/test_images/test_gzip_compressed_image.fits.fz")
array([[0., 0., 0., 0., 0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0., 0., 0., 0., 0.]])
```

Also, all zeros is not good test data. It might not catch edge or corner cases.

**Comment by beckermr** [comment]
frack I somehow got the wrong key or info on the git commit :/

**Comment by erykoff** [comment]
The symptom of the particular bug that I'm testing is that all zeros turns into not-zeros at all.  But I guess I screwed that up because I used the `with fits`.  Fpack compression always moves it to extension 1.

**Comment by beckermr** [comment]
ahh ok. then LGTM!
